pandas/tseries/offsets.py

```diff
@@ -579,9 +580,45 @@ class BusinessHourMixin(BusinessMixin):
 
     def __init__(self, start='09:00', end='17:00', offset=timedelta(0)):
         # must be validated here to equality check
-        start = liboffsets._validate_business_time(start)
+        if _iterable_not_string(start):
+            start = np.asarray(start)
+            if len(start) == 0:
+                raise ValueError('Must include at least 1 start time')
+        else:
+            start = np.array([start])
+        if _iterable_not_string(end):
+            end = np.asarray(end)
+            if len(end) == 0:
+                raise ValueError('Must include at least 1 end time')
+        else:
+            end = np.array([end])
+
+        vliboffsets = np.vectorize(liboffsets._validate_business_time)
+        start = vliboffsets(start)
+        end = vliboffsets(end)
+
+        # Validation of input
+        if len(start) != len(end):
+            raise ValueError('number of starting time and ending time '
+                             'must be the same')
+        num_openings = len(start)
+
+        # sort starting and ending time by starting time
+        index = np.argsort(start)
+        start = tuple(start[index])
```

Comment:
Why do these need to be cast to tuple?

Reply:
So that it's hashable

Reply:
Could you add a comment with that explanation above these?

Reply:
Updated